viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -647,6 +647,23 @@ void SLAMServiceImpl::SaveMapWithTimestamp() {
     }
 }
 
+cartographer::transform::Rigid3d SLAMServiceImpl::AddSensorReading(
+    cartographer::sensor::TimedPointCloudData measurement,
+    cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder,
+    int trajectory_id, cartographer::transform::Rigid3d tmp_global_pose) {
```

Comment:
Per offline discussion:

1. We may need to take in the measurement as a different, C-compatible type in the future, but since we don't know yet what that type will be, we'll leave it as `cartographer::sensor::TimedPointCloudData`

2. We should make the `cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder` and `int trajectory_id` members of SlamServiceImpl, so that they don't need to be passed in here

3. We should refactor the calling code so that `tmp_global_pose` doesn't need to be passed in here